doc/architecture/configuration-and-management.rst

```diff
@@ -0,0 +1,203 @@
+Fragments* are the *REST* resources identified by the paths of the
+actual files. As the *Configuration Fragements* are stored in *JSON* or
+*XML* the *CRUD* operations are trivial.
+
+The *Installer*
+~~~~~~~~~~~~~~~
+
+The task of the\ *Installer* is to generate the actual configuration
+files, typically under /etc, for\ **\ the various *Ostro* software
+components. The *Installer* tracks configuration data changes in the
+*Drop Zone*\ using\ *inotify.*\ When a\ *Configuration
+Fragment*\ changed the *Installer* reads the altered configuration
+values and generates new versions of the effected configuration files
+incorporating the new values. The configuration file generation is
+template based.
+
+A single value change in a *Configuration Fragement* might lead to the
+re-generation of multiple configuration files of various *Ostro*
+components. For instance, if the listening port of the *SSH Daemon* is
+changed, the new value should be incorporated into the
+*/lib/systemd/system/sshd.socket*, */etc/sshd\_config* files as well as
+the *Firewall* rules should be modified.
+
+Each configuration file, that depends on configuration value changes,
+needs to have a dedicated *Template* and a so called *Resource
+Description* file. The *Template* is used to genarete the configuration
+file while the *Resource Description* specifies the context for the
+operation:
+
+-  the resulting configuration file (ie. the output)
+
+-  the template to be used to regenarete the output
+
+-  regeneration triggers (list of what *Configuration Fragment*\ s
+   trigger re-genaration)
+
+-  an optional binary to be invoked to check wheter the generated config
+   file is acceptable or not
+
+-  an optional binary to be invoke to install the new version of
+   configuration, eg. *systemctl reload xxxx.service*
+
+Data model
+----------
+
+The tree of *Configuration Fragements* is composed of three subtrees
+with identical structure:\ *Factory*,\ *Common*\ and\ *Local*.
+
+The *Factory* subtree contains the\ **\ default values of the *Ostro*
+distribution. The *Factory* subtree is part of the *OS* image and is
+immutable.
+
+The *Common* subtree hosts the values that are common for a larger set
+of systems. For instance, a service provider who maintains a large
+number of *Ostro* based systems, could store configuration values under
+the *Common* subtree that are identical for all systems.
+
+The *Local* subtree meant for individual, '*Ostro* box' specific
+configuration values. Configuration values under the *Local* subtree can
+be set either remotely by *Service Provider*\ s or locally by end-users
+using their phones/tablets/laptops. *WLAN SSID* is the typical example
+for locally set configuration data.
+
+*Templates* offer mechanisms to deep-merge these subtrees in increasing
+priority order of *Factory*, *Common* and *Local*.
+
+Components
+----------
+
+The *Ostro Configuration & Management* system is an optional layer. This
```

Comment:
It was not immediately clear that this section switch from "layer in the system" to "OpenEmbedded layer".